lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -36,10 +36,16 @@ class GoogleCloudOutput < BufferedOutput
     config_param :private_key_path, :string, :default => nil
     config_param :private_key_passphrase, :string, :default => 'notasecret'
 
-    # If use_metadata_service is set to true, we obtain the project_id, zone,
-    # and vm_id from the GCE metadata service.  Otherwise, those parameters
-    # must be specified in the config file explicitly.
+    # Specify project/instance metadata.
+    #
+    # project_id, zone, and vm_id are required to have valid values, which
+    # can be obtained from the metadata service or set explicitly.
+    # Otherwise, the plugin will fail to initialize.
+    #
+    # Whether to attempt to obtain metadata from the local metadata service.
+    # It is safe to specify 'true' even on platforms with no metadata service.
     config_param :use_metadata_service, :bool, :default => true
```

Comment:
I don't like the idea of changing the value of one item based on whether other flags are present, that seems confusing.

If we do that, then imagine I'm running on GCE and decide I want to override zone.  Now I can't just say 'zone XXX', I have to also add 'use_metadata_service true' or it will break.

Ideally users will never have to set 'use_metadata_service' regardless of what platform they are running on.